### ISSUE TITLE
VideoPlayer on Android 2.3.3 and Retina display

### ISSUE THREAD

**Issue by minggo** [cocos2d/cocos2d-x#8511]
# Note

 This issue is migrated from [here](http://www.cocos2d-x.org/issues/5709). It was created at 2014/07/11 02:44:48 +0000
# Description

Hi,

I found some issues using VideoPlayer on Android 2.3.3 devices and Retina display.

Issue on Android 2.3.3
I tested VideoPlayer with GalaxyS and GalaxyS2. The OS of these devices is Android 2.3.3.
The video (.mp4) is not played with VideoPlayer of Cocos2d-x 3.3.1.
There is no specific error message in LogCAT.

Issue on Retina display
The video is played well on all iOS simulators. But the scale of video is always 2x on Retina display.

I know VideoPlayer of Cocos2d-x is "Experimental" stage.
But please look into it and fix it.

Thanks and regards,
Junhee
- junhee76 added comment:
  junhee76 wrote:
  > Hi,
  > 
  > I found some issues using VideoPlayer on Android 2.3.3 devices and Retina display.
  > 
  > Issue on Android 2.3.3
  > I tested VideoPlayer with GalaxyS and GalaxyS2. The OS of these devices is Android 2.3.3.
  > The video (.mp4) is not played with VideoPlayer of Cocos2d-x 3.3.1.
  > There is no specific error message in LogCAT.
  > 
  > Issue on Retina display
  > The video is played well on all iOS simulators. But the scale of video is always 2x on Retina display.
  > 
  > I know VideoPlayer of Cocos2d-x is "Experimental" stage.
  > But please look into it and fix it.
  > 
  > Thanks and regards,
  > Junhee

FYI, the mp4 file is well played with internal video player.
- junhee76 added comment:
  Hi,

I found the solution about 2x view on Retina display.

```
--- cocos2d/cocos/ui/UIVideoPlayerIOS.mm
+++ cocos2d/cocos/ui/UIVideoPlayerIOS.mm
@@ -321,7 +321,7 @@
         auto directorInstance = Director::getInstance();
         auto glView = directorInstance->getOpenGLView();
         auto frameSize = glView->getFrameSize();
-        auto scaleFactor = directorInstance->getContentScaleFactor();
+        auto scaleFactor = glView->getContentScaleFactor();

         auto winSize = directorInstance->getWinSize();


```

If one of committers thinks this is the right solution, please apply it with next release.
Thanks and regards,
Junhee

**Comment by minggo** [comment]
@Dhilan007
I think we fixed it, right?

**Comment by dumganhar** [comment]
@Dhilan007, what's status of this issue? Was it fixed?

**Comment by darius2013** [comment]
this bug is also in v3.4 final. my android device is 2.3.4

**Comment by zilongshanren** [comment]
This bug is also in v3.6 final . i9100 can't play local video.

**Comment by walzer** [comment]
Any update?

**Comment by WenhaiLin** [comment]
Fixed can't play video on Android 2.3.x: https://github.com/cocos2d/cocos2d-x/pull/11651
